### PR TITLE
add covers that can hold some items

### PR DESCRIPTION
**Pull request by Glease**
This cover allows player to store a small amount of item alongside a machine. The intended use case is for steam to EV player to store extruder shapes and alloy smelter shapes inside the machine. The cover GUI system would then allow player to retrieve the items right from the machine GUI. To facilitate this goal and not make this cover some generic diamond chest, the current line up impose a 1 item per stack limit. 

itemholdercover

While this demo showcased a 3-slot gui, the submitted PR contains a 3x3 cover instead. The code is structured in a way that the slot count can be tweaked freely without writing any significant code. Multiple cover with different slot count can also be added to game at the same time.

Depends on [this bugfix PR in modular UI](https://github.com/GTNewHorizons/ModularUI/pull/75) to work correctly, but does not require that one to be compiled.

## New items added

Basic Item Holder: 9 slots. Each slot can only hold ONE item no matter what it is
Good Item Holder. 12 slots. Each slot can only hold ONE item no matter what it is
Advanced Item Holder. 15 slots. . Each slot can only hold ONE item no matter what it is

## Screenshots 
gui
in world

## Task list

- [x] basic implementation
- [x] find item texture
- [x] find cover texture (what is displayed when it's placed on a machine)
- [x] determine the tiering
- [x] ensure the meta item id isn't used by something else in fullpack
- [x] try to make the window show up on the side when opened, instead of in the center
- [ ] Add recipes

### CODE REVIEW

File: src/main/java/gregtech/common/covers/GT_Cover_Chest.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    @Override
+    protected void onBaseTEDestroyedImpl(ForgeDirection side, int aCoverID, ChestInventory aCoverVariable,
+        ICoverable aTileEntity) {
+        if (aTileEntity.getWorld().isRemote) return;
+        aCoverVariable.dropAll(aTileEntity, ForgeDirection.UNKNOWN);
+    }
+
+    @Override
+    public ModularWindow createWindow(GT_CoverUIBuildContext buildContext) {
+        return new ChestUIFactory(buildContext).createWindow();
```
Comment (eigenraven) [inline src/main/java/gregtech/common/covers/GT_Cover_Chest.java:91]: Is there a way to force this window to spawn left/right of the main window, so you don't have to drag it around to use it every time?

Comment (Glease) [reply src/main/java/gregtech/common/covers/GT_Cover_Chest.java:91]: this is on the todo list. from what I gathered it *is* possible to place it at a fixed location on the left. The necessary context containing parent window seems to be missing here, so I cannot place it at `max(0, parentWindow.guiLeft - thisWindow.width)`.

**Comment by Dream-Master** [comment]
@Glease feel free to set it undraft when its done.

**Comment by Glease** [comment]
gui
in world

image credit @ wilybit on discord. 
Unfortunately I'm unable to make the cover icon in gui to switch depending on whether the window is open without substantial refactor

**Review by Nxer — CHANGES_REQUESTED** [review]
image

It will open a wrong UI when shift right click the cover side on machine block.

**Comment by Nxer** [comment]
And when break the machine block, the items in cover will drop in world, but re-place the machine(with these covers), there still are items in cover's inventory.

**Comment by YannickMG** [comment]
Only somewhat related: I always wished the crafting table cover held an inventory as that would've made it somewhat useful. As it is it behaves like a vanilla crafting table in that all items you place in it are expelled when you close the UI.

If you think it would make sense to upgrade the crafting table cover to hold an inventory then you might balance this new "chest cover" alongside it.

**Comment by Nxer** [comment]
image
Would it be convenient to add an player inventory UI when opening this UI ?

**Review by Nxer — APPROVED** [review]
Worked well. Good stuff for hand-crafting masters.

**Pull request merged**